### PR TITLE
oc: enable autoscale for dcs

### PR DESCRIPTION
**Pull request by kargakis**
Closes https://github.com/openshift/origin/issues/5947

@fabianofranz @smarterclayton @ncdc @ironcladlou

### CODE REVIEW

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
flaky new-app tests

[test]

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
```
ERROR: gave up waiting for https://127.0.0.1:38443/healthz

!!! Error in /data/src/github.com/openshift/origin/hack/../hack/util.sh:370
	'return 1' exited with status 1
Call stack:
	1: /data/src/github.com/openshift/origin/hack/../hack/util.sh:370 start_os_master(...)
	2: /data/src/github.com/openshift/origin/hack/update-generated-swagger-spec.sh:50 main(...)
Exiting with status 1
```
[test]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/6259 flake

[test]

File: pkg/cmd/cli/cmd/wrappers.go [hunk 1]
```diff
@@ -269,6 +269,29 @@ func NewCmdScale(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Co
 }
 
 const (
+	autoScaleLong = `Creates an autoscaler that automatically chooses and sets the number of pods that run in a kubernetes cluster.
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/wrappers.go]: Make a shorter first line title - move the rest to the second paragraph.  Don't use deploymentConfig, say deployment config.

File: pkg/cmd/cli/kubectl_compat_test.go [hunk 2]
```diff
@@ -17,8 +17,7 @@ import (
 // NEVER ADD TO THIS LIST
 // TODO kill this list
 var MissingCommands = sets.NewString("namespace", "rolling-update", "cluster-info", "api-versions",
-	"apply",     // we don't want to support this implementation
-	"autoscale", // TODO
+	"apply", // we don't want to support this implementation
 )
```
Comment (smarterclayton) [inline pkg/cmd/cli/kubectl_compat_test.go]: Be sure on Rebase you don't reintroduce the apply exclusion

**Comment by smarterclayton** [comment]
A few changes, otherwise Lgtm

File: pkg/cmd/cli/kubectl_compat_test.go (hunk 2 above)
Comment (liggitt) [inline pkg/cmd/cli/kubectl_compat_test.go]: Pretty sure apply got added in the rebase that just landed

File: pkg/cmd/cli/cmd/wrappers.go (hunk 1 above)
Comment (kargakis) [inline pkg/cmd/cli/cmd/wrappers.go]: ok

File: pkg/cmd/cli/kubectl_compat_test.go (hunk 2 above)
Comment (kargakis) [inline pkg/cmd/cli/kubectl_compat_test.go]: rebased on top of latest

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
@smarterclayton any other comments?

**Comment by smarterclayton** [comment]
Lgtm [merge]

**Pull request merged**